zap/src/test/java/org/zaproxy/zap/utils/ZapXmlConfigurationUnitTest.java

```diff
@@ -43,6 +46,92 @@ class ZapXmlConfigurationUnitTest {
+    }
+
+    @Test
+    void shouldSetVariousPropertiesAsExpected()
+            throws ConfigurationException, UnsupportedEncodingException {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        String aString = "Some string 🙂.";
+        String bString = "";
+        String cString = null;
+        int aInt = 120;
+        boolean aBoolean = true;
+        // When
+        config.setProperty("aString", aString);
+        config.setProperty("bString", bString);
+        config.setProperty("cString", cString);
+        config.setProperty("aInt", aInt);
+        config.setProperty("aBoolean", aBoolean);
+        // Then
+        List<Object> aStringList = config.getList("aString");
+        assertThat(aStringList.size(), is(equalTo(1)));
+        assertThat(aStringList.get(0), is(equalTo(aString)));
+
+        List<Object> bStringList = config.getList("bString");
+        assertThat(bStringList.size(), is(equalTo(1)));
+        assertThat(bStringList.get(0), is(equalTo(bString)));
+
+        List<Object> cStringList = config.getList("cString");
+        assertThat(cStringList.size(), is(equalTo(0)));
+
+        List<Object> aIntList = config.getList("aInt");
+        int aActualInt = config.getInt("aInt");
+        assertThat(aIntList.size(), is(equalTo(1)));
+        // Int is string when handled via list
+        assertThat(aIntList.get(0), is(equalTo("120")));
+        assertThat(aActualInt, is(equalTo(120)));
+
+        List<Object> aBooleanList = config.getList("aBoolean");
+        boolean aActualBoolean = config.getBoolean("aBoolean");
+        assertThat(aBooleanList.size(), is(equalTo(1)));
+        // Boolean is string when handled via list
+        assertThat(aBooleanList.get(0), is(equalTo("true")));
+        assertThat(aActualBoolean, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldSetListOfPropertiesAsExpected() {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        String aString = "Some string 🙂.{2,2}";
+        String bString = "Some other string";
+        List<String> aList = List.of(aString, bString);
+        // When
+        config.setProperty("aList", aList);
+        // Then
+        List<Object> readList = config.getList("aList");
+        assertThat(readList.size(), is(equalTo(2)));
+        assertThat(readList.get(0), is(equalTo(aString)));
+    }
+
+    @Test
+    void shouldSetListOfNonPrimitivesAsExpected() throws IOException, ConfigurationException {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        CustomObject coOne = new CustomObject("coOne");
+        CustomObject coTwo = new CustomObject("foo,bar");
+        List<CustomObject> customObjects = List.of(coOne, coTwo);
+        // When
+        config.setProperty("customObjects", customObjects);
+        config.save(outputStream);
```

Comment:
Why `save`?

Reply:
The comma issue only manifests after a save

Reply:
The `List<String>` case is being checked without `save` (and works properly).

Reply:
You're right, that was something else I encountered along the way 😢